### PR TITLE
Fixes #3508 Allow configuration of GoCB opTimeout value

### LINKED ISSUE #3508: Allow configuration of GoCB opTimeout value
## Problems with current behavior

1. Can create an undesirable positive feedback loop where retries trigger more ops which further degrades the performance of the SDK, which causes further timeouts, etc...
1. Increases # of open HTTP request sockets
1. SG API client's (aka CBLite most of the time) HTTP requests can timeout, which will cause t and they implement retry on the client side, which will *exacerbate* the positive feedback loop mentioned above.

## Interim fix

- Allow users to increase the `opTimeout` value that triggers timeouts.  The current default value is 2.5s.  
- Possibly increase the current default value 

This came out of the investigation into the cause of https://github.com/couchbase/sync_gateway/issues/3212

### PR DESCRIPTION
**Pull request by tleyden**
Fixes #3508

### CODE REVIEW

**Comment by tleyden** [comment]
Integration test: (single)

http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-integration-master/547/console

**Review by adamcfraser — COMMENTED** [review]
Looks good - question on the naming of the timeout in the config.

File: rest/config.go [hunk 1]
```diff
@@ -145,6 +145,7 @@ type DbConfig struct {
 	NumIndexReplicasHousekeeping *uint                          `json:"num_index_replicas_housekeeping"`        // Number of GSI index replicas used for housekeeping indexes
 	UseViews                     bool                           `json:"use_views"`                              // Force use of views instead of GSI
 	SendWWWAuthenticateHeader    *bool                          `json:"send_www_authenticate_header,omitempty"` // If false, disables setting of 'WWW-Authenticate' header in 401 responses
+	OperationTimeoutMS           *uint32                        `json:"operation_timeout_ms,omitempty"`         // // How long bucket ops should block returning "operation timed out". If nil, uses GoCB default.  GoCB buckets only.
```
[the remark this answers is not in the document]
Comment (tleyden) [reply rest/config.go]: Yep, definitely agree.  Fixed and pushed up a new commit.

**Review by adamcfraser — APPROVED** [review]

**Comment by tleyden** [comment]
Docs issue: https://issues.couchbase.com/browse/DOC-3552

**Pull request merged**